util/src/filesystem.cpp

```diff
@@ -15,8 +17,36 @@ std::string binaryPathToInstallDir(const char* path)
   // of any symbolic links and relative references.
   // The path is then parented twice (from binary to "bin" folder, and then
   // the root).
-  return fs::canonical(fs::system_complete(fs::path(path)))
-    .parent_path().parent_path().string();
+  // If the path does not exist, the given binary is searched for in the PATH
+  // environment variable.
+
+  if (fs::exists(fs::system_complete(fs::path(path))))
+  {
+    return fs::canonical(fs::system_complete(fs::path(path)))
+      .parent_path().parent_path().string();
+  }
+  else
+  {
+    std::string pPath;
+    std::string fullPath = std::getenv("PATH");
+    std::string delimiter = ":";
+
+    std::size_t pos = 0;
+    while ((pos = fullPath.find(delimiter)) != std::string::npos)
+    {
+      pPath = fullPath.substr(0, pos) + "/" + path;
```

Comment:
`fs::path` has an `operator/` overload for joining paths which uses the delimiter of the current platform. We should use this for joining paths.

Reply:
By the way the separator in `$PATH` can also be queried in some languages. I don't know if this is available in Boost, that's why I havet't requested a change here. But for example Windows uses `;` and unix systems use `:`. Ok, I know we don't support Windows... :)